### ISSUE TITLE
Breaking build because of unawaited

### ISSUE THREAD

**Issue by tigloo** [canonical/dbus_client.dart#304]
It seems that PR #295 broke compilation when removing pedantic as unawaited is now undeclared. Downgrading to 0.6.3 fixes this. Obviously re-adding pedantic again would fix the problem but would reverse the change of PR #295.

```
    ../../.pub-cache/hosted/pub.dartlang.org/dbus-0.6.4/lib/src/dbus_client.dart:791:7: Error: The method 'unawaited' isn't defined for the class 'DBusClient'.
     - 'DBusClient' is from 'package:dbus/src/dbus_client.dart' ('../../.pub-cache/hosted/pub.dartlang.org/dbus-0.6.4/lib/src/dbus_client.dart').
    Try correcting the name to the name of an existing method, or defining a method named 'unawaited'.
          unawaited(_socket!.done.then((value) {
          ^^^^^^^^^
    ../../.pub-cache/hosted/pub.dartlang.org/dbus-0.6.4/lib/src/dbus_server.dart:537:7: Error: The method 'unawaited' isn't defined for the class 'DBusServer'.
     - 'DBusServer' is from 'package:dbus/src/dbus_server.dart' ('../../.pub-cache/hosted/pub.dartlang.org/dbus-0.6.4/lib/src/dbus_server.dart').
    Try correcting the name to the name of an existing method, or defining a method named 'unawaited'.
          unawaited(_processMessage(null, responseMessage));
          ^^^^^^^^^
    ../../.pub-cache/hosted/pub.dartlang.org/dbus-0.6.4/lib/src/dbus_server.dart:1257:5: Error: The method 'unawaited' isn't defined for the class 'DBusServer'.
     - 'DBusServer' is from 'package:dbus/src/dbus_server.dart' ('../../.pub-cache/hosted/pub.dartlang.org/dbus-0.6.4/lib/src/dbus_server.dart').
    Try correcting the name to the name of an existing method, or defining a method named 'unawaited'.
        unawaited(_processMessage(null, message));
        ^^^^^^^^^
```

**Comment by robert-ancell** [comment]
How are you building this? I can do a 'dart test' and 'dart anayze' from the git branch without issues (this is also working on the CI).

**Comment by jpnurmi** [comment]
[unawaited](https://api.dart.dev/dart-async/unawaited.html) is available in dart:async since 2.14

**Comment by tigloo** [comment]
@jpnurmi has the relevant comment. Our Dart version was pinned to 2.13.4 in pubspec.yaml so required pedantic to make it work. Upgrading to a higher version eliminates the problem.

Maybe it makes sense to relabel DBus 0.6.4 as 0.7.0 so it won't break for those including with as ^0.6.0.

Edit: The problem occurred for us because DBus is included as a transitive dependency, we don't control the version. Some package in our dependency tree included obviously with the "^0.6.0" version specifier. So it would break for anyone still on a lower Dart version.

**Comment by robert-ancell** [comment]
https://github.com/canonical/dbus.dart/pull/307 should hopefully work around this issue.
